plugins/net_plugin/net_plugin.cpp

```diff
@@ -1125,10 +1121,10 @@ namespace eosio {
             c->last_handshake_sent.generation = ++c->sent_handshake_count;
             auto last_handshake_sent = c->last_handshake_sent;
             g_conn.unlock();
-            fc_ilog( logger, "Sending handshake generation ${g} to ${ep}, lib ${lib}, head ${head}, id ${id}",
-                     ("g", last_handshake_sent.generation)( "ep", c->peer_name())
-                     ("lib", last_handshake_sent.last_irreversible_block_num)
-                     ("head", last_handshake_sent.head_num)("id", last_handshake_sent.head_id.str().substr(8,16)) );
+            peer_ilog( c, "Sending handshake generation ${g}, lib ${lib}, head ${head}, id ${id}",
```

Comment:
peer_[x]log internally using `conn_mtx` lock which is a bit counterintuitive (one doesn't expect log macro to use class mutex) . In this particular example I see that just before calling peer_ilog `g_conn.unlock` was called. However it looks to me dangerous to rely on this. People can introduce deadlock accidentally.
Documentation says there is a mutex that can be locked inside same thread many times:
`std::recursive_mutex`. Maybe we can switch to it? That way we will protect developers from unintentional causing a deadlock.

Reply:
Google "recursive mutex considered harmful" and enjoy some reading. ;)

Reply:
That said, I was thinking about requiring the `peer_[x]log` to take a mutex so it was obvious. Something like:
`peer_ilog( c, c->conn_mtx, "Sending ...." );` What do you think?

Reply:
ok, will do that, but still I don't like the fact that `peer_[x]log` now becomes in fact default log macro and now we have to release `conn_mtx` every time we want to log some data. Normally I'm risky but in that case it would be nice to have any sort of protection or warning. I can imagine debugging stuck threads in `net_plugin` ...
Any ideas?

Reply:
I agree, hence why I was thinking about passing in the mutex. I have idea of how to remove the mutex, let me give it a try and see how it goes.

Reply:
I think we can restrict all `peer_[x]log` to connection strand and even assert that. Then it should be possible to have access to all connection strand data without a mutex. However, that I think should be a separate PR. I'll revert this commit and create a new PR with just the logging changes.

Reply:
`peer_name()` mutex should be removed at the same time.